src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Helpers.cs

```diff
@@ -137,8 +138,46 @@ public void WriteLine(string source)
                 _sb.AppendLine(source);
             }
 
+            public void WriteBlock(string source)
+            {
+                foreach (string value in source.Split('\n'))
```

Comment:
Nit: you could avoid quite a few intermediate allocations if you replaced `Split` and `Trim` with span-based tokenization.

Reply:
Yeah it would have been lovely if https://github.com/dotnet/runtime/issues/934 was implemented.

Reply:
You might be able to write a bespoke splitter, see

https://github.com/eiriktsarpalis/typeshape-csharp/blob/dcb77f0c94ca762c4c98a6885df117fa4ddcd2a1/src/TypeShape.SourceGenerator/Helpers/SourceWriter.cs#L63-L95

Actually, I think that might be worth the extra effort since source formatting is particularly allocation intensive and avoiding intermediate allocations in a hot-path method would be beneficial.